### DISCUSSION TITLE
[Native DLL] `void*` and `void**`

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#3022] [General]
Use `IntPtr` for both of them? If use `IntPtr`, the declaration is fine. But what about when actually passing data to it?

**Comment by SunSerega** [comment]
`IntPtr` is a wrapper for a pointer that hides what it points to. In the case of `void*` it hides type `void`, in the case of `void**` it hides type `void*`.
You can also translate `void**` as an `array of IntPtr`.
